### PR TITLE
Add star button

### PR DESCRIPTION
**Pull request by vitoriapena**
Issue: #456

### CODE REVIEW

File: assets/js/buttons-examples.js [hunk 1]
```diff
@@ -199,6 +199,12 @@ $(document).ready(function () {
           classes: "scroll-to-top-btn",
           text: "",
         },
+        {
+          type: "star",
+          label: "Star Button",
+          classes: "star-btn",
```
Comment (shahednasser) [inline assets/js/buttons-examples.js:205]: set classes to `""` and add a variation object (you can check social buttons to see how it's done)

File: src/components/icon/_star.less [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    font-family: "Font Awesome 5 Free";
+    display: inline-block;
+    font-style: normal;
+    font-variant: normal;
+    text-rendering: auto;
+    -webkit-font-smoothing: antialiased;
+  }
+
+  &:active {
+    font-weight: 900;
```
Comment (shahednasser) [inline src/components/icon/_star.less:13]: add color yellow on active

**Review by shahednasser — CHANGES_REQUESTED** [review]
It looks good but I think it will look better if the button was just a star without a background. To do this I labeled some changes you need to do in the review below

Let me know if you need any help

**Comment by vitoriapena** [comment]
Thanks for pointing that out, I'll work on it.

File: assets/js/buttons-examples.js (hunk 1 above)
Comment (vitoriapena) [reply assets/js/buttons-examples.js:205]: I added a variation and changed the active color to yellow. But I still have an issue, is it possible to get rid of the rounded and block examples?
image

Comment (shahednasser) [reply assets/js/buttons-examples.js:205]: I need to see what you did for me to let you know. Can you commit your changes?

**Comment by shahednasser** [comment]
Hi @vitoriapena we have added a new way to resolve the issue of rounded-btn and block-btn always being added. You can now add shouldHaveRoundedType to your button object in `buttons` variable and set it to false. shouldHaveBlockType for block buttons as well. 

This PR became kind of messy now though due to merging and resolving conflicts. Can you try to update your forked repo to the latest changes and create a new branch, adding the changes you made? I know it's a hassle but we can't merge a PR like this. Plus you need your repo to be up to date to use the keys above.

I'm closing this PR awaiting a new PR. Thank you!

**Comment by vitoriapena** [comment]
Sure, I'll open a new PR. Thanks!

**Pull request closed without merging**